### ISSUE TITLE
Serialization Currently Misses Base Class / Inherited Properties

### ISSUE THREAD

**Issue by DrewMcArthur** [aloneguid/parquet-dotnet#201]
**Version:** Parquet.Net v4.1.0

**Runtime Version:** .NET6.0

**OS:** Windows

#### Expected behavior

We have a base class with property `A`, and an inherited class with property `B`.  When serializing an instance of the inherited class, we expected the Parquet output to have both properties `A` and `B`.

#### Actual behavior

It appears that the serialize function ignores properties in the base class, so the parquet output only has property `B`.  [We've noticed that this is a noted limitation here](https://github.com/aloneguid/parquet-dotnet/blob/master/docs/serialisation.md#limitations), but this would be nice to have.  Would be happy to work towards this if you could point us in the right direction!

### Steps to reproduce the behavior

1. Create a base class for serialization.
```csharp
public class BaseClass 
{
    [ParquetColumn(Name="property_a")]
    public string PropertyA;
}
```
2. Create another class for serialization, inheriting from the base class.
```csharp
public class InheritedClass : BaseClass
{
    [ParquetColumn(Name="property_b")]
    public string PropertyB;
}
```
3. Create an instance of the inherited class and serialize it.
```csharp
InheritedClass obj = new() 
{
    PropertyA = "a";
    PropertyB = "b";
}
MemoryStream memStream = new();
await ParquetConvert.SerializeAsync(obj, memStream, ...);
```
4.  Check the parquet file, and only see `PropertyB` in the output.  

#### Code snippet reproducing the behavior

I think what I've provided so far is enough to go off of - would be happy to provide more context / a unit test in a PR if you'd like!  Since I found this limitation is mentioned in your documentation, but didn't see any github issues mentioning it, I figured I'd open the issue to track progress.  Thanks for all your work on this library!

**Comment by DrewMcArthur** [comment]
going to start this by opening up a PR with a unit test describing the issue - will link back here once i do 😄

**Comment by DrewMcArthur** [comment]
closed with the merge of #202 ! thanks @aloneguid !
